app/scripts/modules/core/src/pipeline/config/stages/evaluateVariables/EvaluateVariablesStageConfig.tsx

```diff
@@ -0,0 +1,75 @@
+import { IStageConfigProps } from 'core/pipeline/config/stages/core/IStageConfigProps';
+import * as React from 'react';
+import { set } from 'lodash';
+
+import { StageConfigField } from '../core/stageConfigField/StageConfigField';
+import { IStage } from 'core/domain';
+import { MapEditor } from '@spinnaker/core';
+
+export interface IEvaluateVariablesStageConfigState {
+  variables: any;
+}
+
+export class EvaluateVariablesStageConfig extends React.Component<
+  IStageConfigProps,
+  IEvaluateVariablesStageConfigState
+> {
+  public static getDerivedStateFromProps(props: IStageConfigProps): IEvaluateVariablesStageConfigState {
+    const {
+      stage: { variables = [] },
+    } = props;
+    return {
+      variables: EvaluateVariablesStageConfig.compress(variables),
+    };
+  }
+
+  private static compress(variables: any) {
```

Comment:
Lodash has `indexBy`

Reply:
`keyBy` unfortunately uses the entire original object as the value so it would require another transform iteration to pluck.